File: 38-CountandSay/countandSay.java
public class Solution {
    public String countAndSay(int n) {
        String[] rslt = new String[n];
        // if (n == 0) return "";
        rslt[0] = "1";
        for (int i = 1; i < n; i++){
            rslt[i] = "";
            char[] arr = rslt[i-1].toCharArray();
            int j = 0;
            int k = 0;
            while (k < arr.length) {
                while (k < arr.length && arr[k] == arr[j]) k++;
                rslt[i] += (""+(k-j)+arr[j]);
                j = k;
               
            }
        }
        return rslt[n-1];
        
        
        
    }
}

File: 38-CountandSay/countandSay.cpp
class Solution {
public:
    string countAndSay(int n) {
        if(n==0) return "";
        string res="1";
        
        for(int i=2;i<=n;i++){
            string cur="";
            
            for(int j=0;j<res.size();j++){
                int count=1;
                
                while(j+1<res.size()&&res[j+1]==res[j]){
                    count++;
                    j++;
                }
                cur+=to_string(count)+res[j];
            }
            res=cur;
        }
        return res;
    }
};